### PR TITLE
Authentication

### LINKED ISSUE #22: Authentication
Users should be able to login with any valid MySQL user, which gives them access to only those databases/tables they have privileges. Root should be able to access everything.

### PR DESCRIPTION
**Pull request by gilbitron**
Resolves #22 

My attempt at routing authentication to use actual database users. Currently utilises Basic Authentication but this could be replaced by an actual web interface at a later date.

### CODE REVIEW

**Comment by JRGould** [comment]
I feel like kind of a dummy here: I'm not sure how I'm supposed to log in via the basic auth - tried my DB credentials, but that's not working and I can't find anywhere that we would be specifying credentials otherwise?

File: app/Extensions/DatabaseAuthUserProvider.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    /**
+     * Update the "remember me" token for the given user in storage.
+     *
+     * @param  \Illuminate\Contracts\Auth\Authenticatable $user
+     * @param  string $token
+     * @return void
+     */
+    public function updateRememberToken(UserContract $user, $token)
+    {
+        //
```
Comment (gilbitron) [inline app/Extensions/DatabaseAuthUserProvider.php]: This is used for "remember me" tokens. As we don't currently have a web login it doesn't matter if it's empty. If we move to a web based authentication and wanted "remember me" functionality we would need to implement this. I think it's fine as is for just now.

**Comment by gilbitron** [comment]
@JRGould If you create a `.env` file with `APP_DEBUG=true` in it, what does the response return when basic auth fails?

**Comment by JRGould** [comment]
@gilbitron I just get a 401 with data `Invalid Credentials`. 

I tried to trace it through with XDebug and it seems like the methods in `DatabaseAuthUserProvider` never even get executed because it can't seem to find the credentials I typed into the basic auth modal. 

Happy to do a quick screen share on Monday if you want to look into this more.

**Comment by gilbitron** [comment]
@JRGould Strange. I can't think of why this wouldn't work, and it works fine for me. Might need to arrange a screen share at some point unless some else can get it working.

**Comment by JRGould** [comment]
Just spent some time trying to figure this out again in a new environment, I set it up in a Local container, but get the same thing. ¯\_(ツ)_/¯

**Comment by JRGould** [comment]
@gilbitron mind closing this issue? It haunts my mentions...

**Comment by gilbitron** [comment]
@JRGould Sure no problem 👍

**Pull request closed without merging**